scripts/tools/accessControl/SetRolesAccess.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\scripts\tools\accessControl;
+
+use Throwable;
+use oat\oatbox\reporting\Report;
+use oat\oatbox\extension\script\ScriptAction;
+use oat\tao\model\accessControl\func\AclProxy;
+use oat\tao\model\accessControl\func\AccessRule;
+use oat\tao\model\accessControl\ActionAccessControl;
+
+/**
+ * @example php index.php 'oat\tao\scripts\tools\accessControl\SetRolesAccess' [[ --revoke ]] --config [config.json|json_string]
+ */
+class SetRolesAccess extends ScriptAction
+{
+    public const OPTION_REVOKE = 'revoke';
+    public const OPTION_CONFIG = 'config';
+
+    public const CONFIG_RULES = 'rules';
+    public const CONFIG_PERMISSIONS = 'permissions';
+
+    /** @var Report */
+    private $report;
+
+    /** @var bool */
+    private $revoke;
+
+    /** @var array */
+    private $rules;
+
+    /** @var array */
+    private $permissions;
+
+    /**
+     * @example revoke
+     * [
+     *     '--revoke',
+     * ]
+     *
+     * @example config
+     * [
+     *     '--config', [
+     *         'rules' => [
+     *             'role1' => [
+     *                 ['ext' => 'extension', 'mod' => 'controller', 'act' => 'action'],
+     *             ],
+     *             'role2' => [
+     *                 ['ext' => 'extension', 'mod' => 'controller', 'act' => 'action'],
+     *             ],
```

Comment:
Suggestion to improve readability.

```suggestion
     *             'role1' => [
     *                 [
     *                     'ext' => 'extension', 
     *                     'mod' => 'controller', 
     *                      'act' => 'action'
     *                 ],
     *             ],
     *             'role2' => [
     *                 [
     *                     'ext' => 'extension', 
     *                     'mod' => 'controller', 
     *                     'act' => 'action'
     *                 ],
     *             ],
```